src/utils.ts

```diff
@@ -1,13 +1,12 @@
 export const isIterable = (obj: any) => {
-    // checks for null and undefined
-    if (obj == null) {
+    if (!obj) {
```

Comment:
This doesn't return properly for strings. Empty strings are falsy but still valid iterables.

Reply:
Ah true, I'll fix that up later on when I have time

Reply:
In case you're unaware, `obj == null` does a loose equality match for both `null` and `undefined`. If you'd rather be more explicit, you can make a change to `obj === null || obj === undefined` to handle the two options.

This comes down to a matter of principle around using `==` at all in the codebase. This is essentially the only place I will ever use it, unless there is some reason to compare primitives to wrapper classes. (ex: `String("x") === "x"` returns `false` but `String("x") == "x"` returns `true`)

Reply:
[Here](https://2ality.com/2011/12/strict-equality-exemptions.html) is actually a great writeup on it, listing alternatives for different cases where `==` might be used. I think it's worthwhile to move forward with removing all instances of `==` from the codebase.

Reply:
Okay, sounds good. Sorry for disappearing, life heated up. I have some time tonight afaik though, I'll fix it up!